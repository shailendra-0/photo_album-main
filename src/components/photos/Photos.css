.photos-wrapper {
  height: 30vw;
  width: 23vw;
  border: black solid;
  margin: 4px 4px 4px 4px;
}
.photos {
  height: 100%;
  width: 100%;
  object-fit: cover;
}
.photos-wrapper:hover {
  -ms-transform: skewY(2deg); /* IE 9 */
  transform: skewY(2deg);
}
