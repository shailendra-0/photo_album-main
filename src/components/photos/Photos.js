import React from "react";
import "./Photos.css";
const Photos = ({ src, alt = "", className = "", onClick = () => {} }) => {
  return (
    <div className={"photos-wrapper " + className} onClick={onClick}>
      <img className="photos" src={src} alt={alt}></img>
    </div>
  );
};

export default Photos;
