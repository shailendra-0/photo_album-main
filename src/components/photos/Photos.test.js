/* eslint-disable testing-library/no-container */
/* eslint-disable testing-library/no-node-access */
import Photos from "./Photos";
import { fireEvent, render } from "@testing-library/react";

it("renders Photos snapshot", () => {
  const onClick = jest.fn();
  const { asFragment } = render(<Photos src={"#"} onClick={onClick} />);
  expect(asFragment(<Photos src={"#"} onClick={onClick} />)).toMatchSnapshot();
});

it("runs onClick function", () => {
  const onClick = jest.fn();
  const { container } = render(<Photos src={"#"} onClick={onClick} />);
  fireEvent.click(container.querySelector("img"));
  expect(onClick).toHaveBeenCalled();
});
