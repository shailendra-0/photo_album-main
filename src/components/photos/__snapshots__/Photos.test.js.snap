// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Photos snapshot 1`] = `
<DocumentFragment>
  <div
    class="photos-wrapper "
  >
    <img
      alt=""
      class="photos"
      src="#"
    />
  </div>
</DocumentFragment>
`;
